src/main/java/org/zaproxy/zap/extension/jwt/utils/JWTConstants.java

```diff
@@ -31,13 +31,14 @@
 
     char JWT_TOKEN_PERIOD_CHARACTER = '.';
     String JWT_TOKEN_PERIOD_CHARACTER_REGEX = "[" + JWT_TOKEN_PERIOD_CHARACTER + "]";
-    Pattern JWT_TOKEN_REGEX_PATTERN =
-            Pattern.compile(
-                    "[a-zA-Z0-9_-]*"
-                            + JWT_TOKEN_PERIOD_CHARACTER_REGEX
-                            + "[a-zA-Z0-9_-]*"
-                            + JWT_TOKEN_PERIOD_CHARACTER_REGEX
-                            + "[a-zA-Z0-9_-]*$");
+    String JWT_TOKEN_FORMAT_REGEX =
+            "[a-zA-Z0-9_-]*"
+                    + JWT_TOKEN_PERIOD_CHARACTER_REGEX
+                    + "[a-zA-Z0-9_-]*"
+                    + JWT_TOKEN_PERIOD_CHARACTER_REGEX
+                    + "[a-zA-Z0-9_-]*";
+    // Pattern JWT_TOKEN_REGEX_VALIDATOR_PATTERN = Pattern.compile(JWT_TOKEN_FORMAT_REGEX + "$");
+    Pattern JWT_TOKEN_REGEX_PATTERN = Pattern.compile(JWT_TOKEN_FORMAT_REGEX);
     String BASE64_PADDING_CHARACTER_REGEX = "[=]";
     String[] NONE_ALGORITHM_VARIANTS = {"none", "None", "NONE", "nOnE"};
```

Comment:
This isn't from this PR but it stuck out while I was scrolling. `"NoNe"` and other possible leading/end caps combos are missing. Couldn't a case-insensitive comparison be done wherever it's used? (Feel free to tell me if I'm wrong, I don't recall where/how it's used and didn't go digging.)

Reply:
Actually these combinations go as an input to vulnerable Application like
`
{
 "alg" : "None"
}
`
I am not sure should we try all the character combinations of NONE or not. I will read about it but in case applications are stoping "NONE" only but not other combinations then we can succeed to do this attack.